c2cgeoportal/tests/functional/test_themes_edit_columns.py

```diff
@@ -0,0 +1,395 @@
+# either expressed or implied, of the FreeBSD Project.
+
+
+from nose.plugins.attrib import attr
+from unittest import TestCase
+
+import sqlahelper
+from c2cgeoportal.tests.functional import (  # noqa
+    tear_down_common as tearDownModule,
+    set_up_common as setUpModule,
+    create_dummy_request,
+    create_default_ogcserver,
+)
+
+
+@attr(functional=True)
+class TestThemesEditColumns(TestCase):
+
+    _table_index = 0
+
+    def setUp(self):
+        # Always see the diff
+        # https://docs.python.org/2/library/unittest.html#unittest.TestCase.maxDiff
+        self.maxDiff = None
+        self._tables = []
+
+        import transaction
+        from c2cgeoportal.models import DBSession, Role, User, Interface, TreeItem, Theme, \
+            LayerGroup, OGCSERVER_AUTH_NOAUTH
+        from c2cgeoportal.lib.dbreflection import init
+
+        for treeitem in DBSession.query(TreeItem).all():
+            DBSession.delete(treeitem)
+
+        self.role = Role(name=u"__test_role")
+        self.user = User(
+            username=u"__test_user",
+            password=u"__test_user",
+            role=self.role
+        )
+        self.main = Interface(name=u"main")
+
+        self.ogc_server, external_ogc_server = create_default_ogcserver()
+        self.ogc_server.auth = OGCSERVER_AUTH_NOAUTH
+        external_ogc_server.auth = OGCSERVER_AUTH_NOAUTH
+
+        self.metadata = None
+        self.layer_ids = []
+
+        self.layer_group_1 = LayerGroup(name=u"__test_layer_group_1")
+
+        theme = Theme(name=u"__test_theme")
+        theme.interfaces = [self.main]
+        theme.children = [self.layer_group_1]
+
+        DBSession.add_all([self.main, self.user, self.role, theme, self.layer_group_1])
+
+        transaction.commit()
+
+        engine = sqlahelper.get_engine()
+        init(engine)
+
+    def tearDown(self):  # noqa
+        import transaction
+        from c2cgeoportal.models import DBSession, Role, User, TreeItem, \
+            RestrictionArea, Interface, OGCServer
+
+        transaction.commit()
+
+        for treeitem in DBSession.query(TreeItem).all():
+            DBSession.delete(treeitem)
+
+        DBSession.query(User).filter(
+            User.username == u"__test_user"
+        ).delete()
+        r = DBSession.query(Role).filter(
+            Role.name == u"__test_role"
+        ).one()
+        r.restrictionareas = []
+        DBSession.delete(r)
+        DBSession.query(RestrictionArea).filter(
+            RestrictionArea.name == u"__test_ra"
+        ).delete()
+
+        DBSession.query(Interface).filter(
+            Interface.name == u"main"
+        ).delete()
+
+        DBSession.query(OGCServer).delete()
+
+        for table in self._tables[::-1]:
+            table.drop(checkfirst=True)
+
+        transaction.commit()
+
+    def _create_layer(
+            self, public=False, none_area=False, attr_list=False,
+            exclude_properties=False, metadatas=None, geom_type=False):
```

Comment:
public is always False, none_area is always False, attr_listis always False, then they should be removed ...